src/components/FormSelectList.vue

```diff
@@ -50,390 +51,414 @@
-            return;
-          }
+      const fieldName = this.options.key || "value";
 
-          let params = {
-            config: {
-              endpoint: selectedEndPoint,
-            }
-          }
+      return this.stripMustache(fieldName);
+    },
+    optionsValue() {
+      if (
+        this.options.dataSource &&
+        this.options.dataSource === "provideData"
+      ) {
+        return "content";
+      }
+      return "__content__";
+    },
+    classList() {
+      return {
+        "has-errors":
+          (this.validator && this.validator.errorCount) || this.error,
+        [this.controlClass]: !!this.controlClass
+      };
+    }
+  },
+  mounted() {
+    // reset the value to null if the options list does not contain the selected value
+    // Special Case String Value:
+    //   Review test tests/e2e/specs/MultiselectWithStringValue.spec.js in ScreenBuilder
+    const resetValueIfNotInOptions = typeof this.value !== "string";
+    this.fillSelectListOptions(resetValueIfNotInOptions);
+  },
+  methods: {
+    async loadOptionsFromDataConnector(options) {
+      const { selectedEndPoint } = options;
+      const { selectedDataSource } = options;
+      const { dataName } = options;
 
-          if (typeof this.options.pmqlQuery !== 'undefined' && this.options.pmqlQuery !== '' && this.options.pmqlQuery !== null) {
-            const pmql = Mustache.render(this.options.pmqlQuery, {data: this.validationData});
-            params.config.outboundConfig = [
-              { type: 'PARAM', key: 'pmql', value: pmql }
-            ];
-          }
+      // If no data source has been specified, do not make the api call
+      if (
+        selectedDataSource === null ||
+        typeof selectedDataSource === "undefined" ||
+        selectedDataSource.toString().trim().length === 0
+      ) {
+        return false;
+      }
 
-          // Do not re-run the same request
-          const request = { selectedDataSource, params };
-          if (isEqual(this.lastRequest, request)) {
-            return;
-          }
-          this.lastRequest = cloneDeep(request);
-          
-          this.$dataProvider.postDataSource(selectedDataSource, null, params)
-              .then(response => {
-                const list = dataName ? eval('response.data.' + dataName) : response.data;
+      // Do not run in standalone mode
+      if (!this.$dataProvider) {
+        return false;
+      }
 
-                const transformedList = this.transformOptions(list);
-                this.$root.$emit('selectListOptionsUpdated', transformedList);
-                this.selectListOptions = transformedList;
-              })
-              .catch(err => {
-                /* Ignore error */
-              });
-        }, 700)
+      // If no endpoint has been specified, do not make the api call
+      if (
+        selectedEndPoint === null ||
+        typeof selectedEndPoint === "undefined" ||
+        selectedEndPoint.toString().trim().length === 0
```

Comment:
this could be refactored to
```suggestion
!!selectedDataSource && selectedDataSource.toString().trim().length === 0
```

Reply:
done but with selectedEndPoint variable